<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>923</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="main_widget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QSplitter" name="main_splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QTabWidget" name="info_tabs">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <property name="movable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="physics_tab">
        <property name="isMovable" stdset="0">
         <bool>true</bool>
        </property>
        <attribute name="title">
         <string>Physics</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="ms_per_step_label">
           <property name="text">
            <string>Time Per Step (ms)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="ms_per_step_spinner">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="real_time_factor_label">
           <property name="text">
            <string>RealTime Factor</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="real_time_factor_spinner">
           <property name="decimals">
            <number>3</number>
           </property>
           <property name="maximum">
            <double>10.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.050000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="mouse_tab">
        <attribute name="title">
         <string>Mouse</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QGroupBox" name="control_group">
           <property name="title">
            <string>Control</string>
           </property>
           <layout class="QFormLayout" name="formLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="fwd_label">
              <property name="text">
               <string>Forward (cps)</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="fwd_spinbox">
              <property name="minimum">
               <number>-255</number>
              </property>
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="singleStep">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="turn_label">
              <property name="text">
               <string>Turn (deg/s)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="turn_spinbox">
              <property name="minimum">
               <number>-255</number>
              </property>
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="singleStep">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="teleport_column_label">
              <property name="text">
               <string>Column</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QDoubleSpinBox" name="teleport_column_spinbox">
              <property name="decimals">
               <number>5</number>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.012000000000000</double>
              </property>
              <property name="value">
               <double>0.500000000000000</double>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="teleport_row_label">
              <property name="text">
               <string>Row</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QDoubleSpinBox" name="teleport_row_spinbox">
              <property name="decimals">
               <number>5</number>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.012000000000000</double>
              </property>
              <property name="value">
               <double>0.500000000000000</double>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QLabel" name="teleport_yaw_label">
              <property name="text">
               <string>Yaw</string>
              </property>
             </widget>
            </item>
            <item row="7" column="1">
             <widget class="QDoubleSpinBox" name="teleport_yaw_spinbox">
              <property name="decimals">
               <number>5</number>
              </property>
              <property name="minimum">
               <double>-3.141590000000000</double>
              </property>
              <property name="maximum">
               <double>3.141590000000000</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
             </widget>
            </item>
            <item row="8" column="0">
             <widget class="QPushButton" name="teleport_button">
              <property name="text">
               <string>Teleport</string>
              </property>
             </widget>
            </item>
            <item row="9" column="0">
             <widget class="QCheckBox" name="static_checkbox">
              <property name="toolTip">
               <string>Wheels will move but the robot will not.</string>
              </property>
              <property name="text">
               <string>Static</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QPushButton" name="send_command_button">
              <property name="text">
               <string>Send Cmd</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="maze_tab">
        <attribute name="title">
         <string>Maze</string>
        </attribute>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QPushButton" name="load_maze_button">
           <property name="text">
            <string>Load Maze File</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="edit_baze_button">
           <property name="text">
            <string>Edit Maze</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QPushButton" name="random_maze_button">
           <property name="text">
            <string>Random Maze</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="maze_file_name_label">
           <property name="text">
            <string>No Maze</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="editing_label">
           <property name="text">
            <string>Not Editing</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="pid_tab">
        <attribute name="title">
         <string>PID Control</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QGroupBox" name="constants_group">
           <property name="title">
            <string>Constants</string>
           </property>
           <layout class="QFormLayout" name="formLayout_5">
            <item row="5" column="0">
             <widget class="QPushButton" name="publish_constants_button">
              <property name="text">
               <string>Publish</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="kp_label">
              <property name="text">
               <string>kP</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="kp_spinbox">
              <property name="minimum">
               <double>-9999.000000000000000</double>
              </property>
              <property name="maximum">
               <double>9999.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="ki_label">
              <property name="text">
               <string>kI</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="ki_spinbox">
              <property name="minimum">
               <double>-99.989999999999995</double>
              </property>
              <property name="singleStep">
               <double>0.050000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="kd_label">
              <property name="text">
               <string>kD</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="kd_spinbox">
              <property name="minimum">
               <double>-99.989999999999995</double>
              </property>
              <property name="singleStep">
               <double>0.050000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="kff_offset_label">
              <property name="text">
               <string>kFF Offset</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QDoubleSpinBox" name="kff_offset_spinbox">
              <property name="minimum">
               <double>-99.989999999999995</double>
              </property>
              <property name="singleStep">
               <double>0.050000000000000</double>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="kff_scale_label">
              <property name="text">
               <string>kFF Scale</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QDoubleSpinBox" name="kff_scale_spinbox">
              <property name="minimum">
               <double>-9999.000000000000000</double>
              </property>
              <property name="maximum">
               <double>9999.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="setpoints_group">
           <property name="title">
            <string>Setpoints</string>
           </property>
           <layout class="QFormLayout" name="formLayout_6">
            <item row="2" column="0">
             <widget class="QPushButton" name="publish_setpoints_button">
              <property name="text">
               <string>Publish</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="left_setpoint_label">
              <property name="text">
               <string>Left Speed (c/s)</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="left_setpoint_spinbox">
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="right_setpoint_label">
              <property name="text">
               <string>Right Speed (c/s)</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="right_setpoint_spinbox">
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="right_side_layout">
        <item>
         <widget class="QTabWidget" name="gui_tabs">
          <property name="currentIndex">
           <number>-1</number>
          </property>
          <property name="tabsClosable">
           <bool>false</bool>
          </property>
          <property name="movable">
           <bool>false</bool>
          </property>
          <property name="tabBarAutoHide">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="time_status_layout">
      <item>
       <widget class="QPushButton" name="step_button">
        <property name="text">
         <string>Step</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="step_size_label">
        <property name="text">
         <string>Step Size</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="step_spinner">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1000</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="play_button">
        <property name="text">
         <string>Play</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="real_time_label">
        <property name="text">
         <string>RTF:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="real_time_value_label">
        <property name="text">
         <string>1.0</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="time_label">
        <property name="text">
         <string>Time:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="time_value_label">
        <property name="text">
         <string>00:00:00.000</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>923</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
    <addaction name="actionRestart"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionSourceCode"/>
    <addaction name="actionWiki"/>
    <addaction name="actionKeyboard_Shortcuts"/>
   </widget>
   <widget class="QMenu" name="menuWorld">
    <property name="title">
     <string>Server</string>
    </property>
    <addaction name="actionReset_Mouse"/>
    <addaction name="actionReset_Time"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuWorld"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionSourceCode">
   <property name="text">
    <string>Source Code</string>
   </property>
  </action>
  <action name="actionDebug_Console">
   <property name="text">
    <string>Debug Console</string>
   </property>
  </action>
  <action name="actionReset_Mouse">
   <property name="text">
    <string>Reset Mouse</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionReset_Time">
   <property name="text">
    <string>Reset Time</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
  <action name="actionWiki">
   <property name="text">
    <string>Wiki</string>
   </property>
  </action>
  <action name="actionRestart">
   <property name="text">
    <string>Restart</string>
   </property>
  </action>
  <action name="actionKeyboard_Shortcuts">
   <property name="text">
    <string>Keyboard Shortcuts</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="simulator.qrc"/>
 </resources>
 <connections/>
</ui>
